syntax = "proto3";

enum LogLevel {
  UNKNOWN = 0;
  DEBUG = 1;
  INFO = 2;
  WARNING = 3;
  ERROR = 4;
  CRITICAL = 5;
}

message LogRecord {
  LogLevel level = 1;
  uint32 tick = 2;
  uint64 hash = 3;

  message Arg {
    oneof value {
      uint32 u32 = 1;
      int32 i32 = 2;
      float f32 = 3;
    }
  }
  repeated Arg args = 4;
}
